components/SerialServer/src/serial.c

```diff
@@ -404,16 +404,16 @@ static void handle_char(uint8_t c)
             statemachine = 1;
             break;
         default:
+            last_out = -1;
+            statemachine = 1;
             if (c >= '0' && c < '0' + num_getchar_clients) {
-                last_out = -1;
                 int client = c - '0';
                 printf(COLOR_RESET "\r\nSwitching input to %d\r\n", client);
                 active_client = client;
-                statemachine = 1;
             } else {
-                statemachine = 0;
-                give_client_char(ESCAPE_CHAR);
-                give_client_char(c);
+                printf(COLOR_RESET "\r\nInvalid SerialServer command: %c"
+                       "\r\nType %c? for help"
+                       "\r\n", (char)c, ESCAPE_CHAR);
```

Comment:
If someone was trying to type `@foo` into a client, then the previous behavior would be that the client would see `@foo`.  The new behavior appears that the client will now see `oo` right? 

If a user now wanted the client to see `@foo` then they would need to input `@@foo`.  Does it make sense that this should be the only way to send strings starting with the `ESCAPE_CHAR` to the client? This would be safer for extending the serial server commands in the future as it would reserve all characters for possible serial server commands.

Reply:
Correct. You'd have to double-type the `ESCAPE_CHAR` (Only after the user has entered a newline). I think the benefits outweigh the drawbacks, considering its an edge-case.